### ISSUE TITLE
NTLM Proxy-Authentication does not work when responseHeaders are set in callbacks.

### ISSUE THREAD

**Issue by pcad** [electron/electron#13427]
* Electron Version: 1.7.13, 1.8.7 
* Operating System (Platform and Version): Windows 
* Last known working Electron version: N/A

**Expected Behavior**
NTLM Proxy-Authentication should work automatically for users.

**Actual behavior**
When responseHeaders are set in an onHeadersReceived callback, authorization and authentication is not seen in the logs, and the page does not successfully navigate.

When no headers are set in the callback and a simple `callback({cancel:false})` is executed, the proxy authentication is successfully established and web-traffic works as expected.

**To Reproduce**
This is difficult to reproduce and requires an environment with a proxy.  In our environment, we are running a McAffee proxy that delivers a PAC file to the client.  The client then attempts to make a connection to a service and the proxy requires NTLM authentication.

**Test App**
```js
const {app, BrowserWindow, session} = require('electron');

app.on('window-all-closed', () => {
  app.quit();
});

app.on('ready', () => {

  const browserWindow = new BrowserWindow({
    title: 'app',
    height: 768,
    width: 1024
  });

  browserWindow.loadURL('https://google.com/');


  function onNavigatorHeadersReceived(details, callback){
    callback({cancel:false, responseHeaders: details.responseHeaders});
  }

  session.defaultSession.webRequest.onHeadersReceived({urls: ['*']}, onNavigatorHeadersReceived);
});
```

**Additional Information**
This issue is *not* found in electron 2.0.3.

I've put together a gist to detail the HTTPS conversation.  

https://gist.github.com/pcad/b51fe11d99f4c0fcd89d98e3dba90aa1

Using wireshark I [captured a Chrome session](https://gist.github.com/pcad/b51fe11d99f4c0fcd89d98e3dba90aa1#file-successful-wireshark-log-txt) where the browser sets headers for `Proxy-Authorization` and the proxy responds with a `Proxy-Authenticate` that includes the NTLM hashes that allow the conversation to successfully complete.  

Because this conversation is happening over TLS, I am not able to get an unencrypted stream for an electron session.  Instead, I have dumped the headers from `onBeforeSendHeaders ` and `onHeadersReceived`. In [the unsuccessful navigation](https://gist.github.com/pcad/b51fe11d99f4c0fcd89d98e3dba90aa1#file-blank-page-txt-L13),  the proxy responds twice with the `Proxy-Authenticate` payload of `["NTLM", "Basic realm=\"examplehealth\""],`.  The NTLM hashes that are found in the successful `Proxy-Authenticate` are never seen.

I have tried this with and without `ses.allowNTLMCredentialsForDomains('*')` and was unable to successfully get authenticated.

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v2.0.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons.

Thanks in advance! Your help is appreciated.

**Comment by sofianguy** [comment]
Thank you for your issue!

We haven't gotten a response to our questions in our comment above. With only the information that is currently in the issue, we don't have enough information to take action. I'm going to close this but don't hesitate to reach out if you have or find the answers we need, we'll be happy to reopen the issue.

**Comment by enahum** [comment]
@pcad were you able to respond to the NTLM challenge successfully? I've been looking for a solution everywhere and cannot find anything useful.

**Comment by pcad** [comment]
@enahum  - Are you setting headers in onHeadersReceived?  

iirc, we were able to work around this issue by calling the callback with a {cancel:false} instead of {cancel:false, responseHeaders:{...}}.  Again, just going off of memory, there were some headers that were blocked from being passed through if we passed through any headers.  By eliminating the responseHeaders from the callback, the original headers were used.

Are you on 1.x?  I think this went away with later versions.
